app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1191,6 +1234,209 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
         }
     }
 
+    inner class BrowserTabFragmentExperimentDecorator {
+        fun decorateToolbarMenus(viewState: BrowserViewState) {
+            if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.BottomBarNavigationExperiment)) {
+                decorator.hideToolbarMenu()
+            } else {
+                decorator.decorateToolbarMenu(viewState)
+            }
+        }
+
+        private fun decorateToolbarMenu(viewState: BrowserViewState) {
+            tabsButton?.isVisible = viewState.showTabsButton
+            fireMenuButton?.isVisible = viewState.showFireButton
+            menuButton?.isVisible = viewState.showMenuButton
+        }
+
+        private fun hideToolbarMenu() {
+            menuButton?.gone()
+        }
+
+        fun decorateWithToolbarOnlyExperiment() {
+            hideBottomBar()
+            decorateAppBarWithToolbarOnlyExperiment()
+            createPopupMenuWithToolbarOnlyExperiment()
+            configureShowTabSwitcherListenerWithToolbarOnlyExperiment()
+            configureLongClickOpensNewTabListenerWithToolbarOnlyExperiment()
+        }
+
+        fun decorateWithBottomBarSearchExperiment() {
+            decorateBottomBarWithNavigationOnlyExperiment()
+            decorateAppBarWithBottomBarExperiment()
+            createPopupMenuWithBottomBarExperiment()
+            configureShowTabSwitcherListenerWithBottomBarNavigationOnlyExperiment()
+            configureLongClickOpensNewTabListenerWithBottomBarNavigationOnlyExperiment()
+        }
+
+        private fun decorateAppBarWithToolbarOnlyExperiment() {
+            toolbar.inflateMenu(R.menu.menu_browser_activity)
+            toolbar.setOnMenuItemClickListener { menuItem ->
+                when (menuItem.itemId) {
+                    R.id.fire -> {
+                        browserActivity?.launchFire()
+                        pixel.fire(String.format(Locale.US, Pixel.PixelName.MENU_ACTION_FIRE_PRESSED.pixelName, variantManager.getVariant()))
+                        return@setOnMenuItemClickListener true
+                    }
+                    else -> return@setOnMenuItemClickListener false
+                }
+            }
+        }
+
+        private fun decorateAppBarWithBottomBarExperiment() {
+            menuButton?.gone()
+        }
+
+        private fun createPopupMenuWithToolbarOnlyExperiment() {
+            popupMenu = BrowserPopupMenu(layoutInflater, variantManager.getVariant())
+            val view = popupMenu.contentView
+            popupMenu.apply {
+                onMenuItemClicked(view.forwardPopupMenuItem) { viewModel.onUserPressedForward() }
+                onMenuItemClicked(view.backPopupMenuItem) { activity?.onBackPressed() }
+                onMenuItemClicked(view.refreshPopupMenuItem) {
+                    viewModel.onRefreshRequested()
+                    pixel.fire(String.format(Locale.US, Pixel.PixelName.MENU_ACTION_REFRESH_PRESSED.pixelName, variantManager.getVariant()))
```

Comment:
do we need that `Locale.US` here?

Reply:
it throws a lint warning, and by default we should always apply a Locale when formatting a String. Since the pixel names are in English, this Locale is the appropriate to use